session/session.go

```diff
@@ -2843,30 +2844,77 @@ func logStmt(execStmt *executor.ExecStmt, s *session) {
 	}
 }
 
-func logGeneralQuery(execStmt *executor.ExecStmt, s *session, isPrepared bool) {
+func genFnGetQuery(execStmt *executor.ExecStmt, s *session, isPrepared bool) func(buf *strings.Builder) string {
 	vars := s.GetSessionVars()
```

Comment:
Is it possible that session vars get changed after the execution, which will then causing the async flow to get wrong values?